### DISCUSSION TITLE
Anaconda create partitions on the disk in random order, but not in the order described in the kickstart file

### DISCUSSION THREAD

**Discussion by makhomed** [rhinstaller/anaconda#4675] [Q&A] [answered]
```
Name         : anaconda
Version      : 34.25.1.14
Release      : 1.el9.rocky.0.3
```

Reporting anaconda issues via github is blocked - how it is possible to report anaconda bugs to anaconda developers?

In the kickstart file I describe order of creation partitions on the disks, but anaconda create partitions in random order. 

fragment of kickstart file:
```
ignoredisk --only-use=nvme0n1,nvme1n1
zerombr
clearpart --all --initlabel --disklabel=gpt --drives=nvme0n1,nvme1n1
bootloader --location=mbr --timeout=1

part biosboot.11 --fstype=biosboot --size=1 --ondisk=nvme0n1
part biosboot.12 --fstype=biosboot --size=1 --ondisk=nvme1n1

part raid.21 --size=1024 --ondisk=nvme0n1
part raid.22 --size=1024 --ondisk=nvme1n1
raid /boot --label=boot --fstype=ext4 --fsoptions="defaults,noatime,discard" --level=1 --device=boot raid.21 raid.22

part raid.31 --size=295044 --ondisk=nvme0n1
part raid.32 --size=295044 --ondisk=nvme1n1
raid pv.41 --level=1 --device=pv raid.31 raid.32

volgroup rocky pv.41
logvol swap  --vgname=rocky --name=swap --label=swap --fstype=swap --fsoptions="defaults,noatime,discard" --size=262144
logvol /     --vgname=rocky --name=root --label=root --fstype=ext4 --fsoptions="defaults,noatime,discard" --size=8192
logvol /home --vgname=rocky --name=home --label=home --fstype=ext4 --fsoptions="defaults,noatime,discard" --size=8192
logvol /opt  --vgname=rocky --name=opt  --label=opt  --fstype=ext4 --fsoptions="defaults,noatime,discard" --size=8192
logvol /var  --vgname=rocky --name=var  --label=var  --fstype=ext4 --fsoptions="defaults,noatime,discard" --size=8192
```

expected order:
```
1. biosboot
2. /boot
3.  swap
4. /
5. /home
6. /opt
7. /var
```
actual order:
```
1. biosboot
2. /
3. swap
4. /var
5. /opt
6. /home
7. /boot
```

```
# lsblk
NAME             MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINTS
nvme0n1          259:0    0   3.5T  0 disk
├─nvme0n1p1      259:2    0     1M  0 part
├─nvme0n1p2      259:3    0 288.1G  0 part
│ └─md127          9:127  0   288G  0 raid1
│   ├─rocky-root 253:0    0     8G  0 lvm   /
│   ├─rocky-swap 253:1    0   256G  0 lvm   [SWAP]
│   ├─rocky-var  253:2    0     8G  0 lvm   /var
│   ├─rocky-opt  253:3    0     8G  0 lvm   /opt
│   └─rocky-home 253:4    0     8G  0 lvm   /home
└─nvme0n1p3      259:4    0     1G  0 part
  └─md126          9:126  0  1022M  0 raid1 /boot
nvme1n1          259:1    0   3.5T  0 disk
├─nvme1n1p1      259:5    0     1M  0 part
├─nvme1n1p2      259:6    0 288.1G  0 part
│ └─md127          9:127  0   288G  0 raid1
│   ├─rocky-root 253:0    0     8G  0 lvm   /
│   ├─rocky-swap 253:1    0   256G  0 lvm   [SWAP]
│   ├─rocky-var  253:2    0     8G  0 lvm   /var
│   ├─rocky-opt  253:3    0     8G  0 lvm   /opt
│   └─rocky-home 253:4    0     8G  0 lvm   /home
└─nvme1n1p3      259:7    0     1G  0 part
  └─md126          9:126  0  1022M  0 raid1 /boot
``

**Comment by VladimirSlavik** [comment, marked as answer]
> how it is possible to report anaconda bugs to anaconda developers

1. Click this link: https://bugzilla.redhat.com/enter_bug.cgi?product=Fedora&component=anaconda&version=rawhide
2. Pick the right _Version_
3. Fill in at least _Summary_ and _Details_
4. After you submit the bug, attach to it also logs from `/tmp` that anaconda writes. In most cases, we need `journal.log` and/or `syslog` ones as a bare minimum. See also https://anaconda-installer.readthedocs.io/en/latest/common-bugs.html#missing-logs

----

Note that the linked bug tracker is for Fedora. There's also RHEL, if you start with the "New" link and pick Red Hat products, and CentOS Stream is a version under RHEL. If you use anaconda on some other distro, it's likely that the anaconda is 1:1 with some distro we support, but we don't really have the capacity to trust that or verify that. Please report only problems reproduced on distros we support.

**Comment by VladimirSlavik** [comment]
> In the kickstart file I describe order of creation partitions on the disks, but anaconda create partitions in random order.

The order is not kept.

As far as I can tell, we never promise this anywhere. However, it's not advertised that it might differ from the input, either, which is arguably a mistake.

Technically, this is something blivet, our storage library, decides. We give it the available storage, the requests, and it does the whole magic. The new layout must take into account existing partitions, required order of things etc., so user-specified order is the one constraint that is not offered at all.

If you really need a specific order of partitions, you can create the partitioning either manually, or in a `%pre` section, and `mount` the results instead.
